packages/runtime/src/builder/serialization/control-serialization.ts

```diff
@@ -866,6 +870,52 @@ function deserializeRichTextControl(
   return { ...serializedControl, options: deserializedOptions }
 }
 
+type RichTextV2PluginControlDefinition<T extends ControlDefinition> = {
+  definition: T
+}
+
+type RichTextV2Plugin = {
+  controls?: RichTextV2PluginControlDefinition<ControlDefinition>[]
```

Comment:
I think we should change this to just `control` or `type` and if you want to compose multiple controls, you can use `Shape`. What do you think?

Reply:
Let's say you create a plugin with `Shape` (or same principle goes for `List`)
```
export const BlockPlugin = createRichTextV2Plugin({
  withPlugin: withBlock,
  controls: Shape({
    type: {
      foo: {
        definition: Select({
          label: 'Block',
          options: [
            {
              value: BlockType.Heading1,
              label: 'Heading1',
            },
            /* ... */
            {
              value: BlockType.Paragraph,
              label: 'Paragraph',
            },
          ],
          defaultValue: BlockType.Paragraph,
      },
      onChange: () => {},
      getValue: () => {}
    bar: // Same thing
```

What do the keys (`foo` and `bar`) map to? There are no props or schema that we are changing. 

The reason I designed it as I did it is that these are stateless ways of mutating whatever is currently selected.  

We do need them to have a way of specifying multiple controls because a plugin isn't 1on1 with a control.

Reply:
After some more thought. Our current plugins are only going to use one control per plugin. I might just move down to 
```suggestion
  controls?: RichTextV2PluginControlDefinition<ControlDefinition>
```
because I am really struggling with how `List` and `Shape` would compose `getValue` / `onChange`

Reply:
What you have above is not how the Shape control’s API works. You just have a value and on change per plugin, not per control. In fact, what you said in your second comment is what I meant. One control per plugin is fine, of you need a group of values you can use List or Shape